src/core/openthread-core-default-config.h

```diff
@@ -914,10 +914,20 @@
 #define OPENTHREAD_CONFIG_NUM_DHCP_PREFIXES 4
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_ENABLE_SLAAC
+ *
+ * Define as 1 to enable support for adding of auto-configured SLAAC addresses by OpenThread.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_ENABLE_SLAAC
+#define OPENTHREAD_CONFIG_ENABLE_SLAAC 0
```

Comment:
Generally I agree that it makes sense to disable SLAAC by default (it follows how we did other optional features). However from Openthread users perspective I already imagine many potential issues and debugging time due to this default behaviour. Mostly because previously both NCP and CLI examples enabled SLAAC by default. Thoughts on setting this variable to 1?

Reply:
@LuDuda , the relatively recent PR https://github.com/openthread/openthread/pull/3484 brought the SLAAC feature support into core. Before this CLI build enabled support for SLAAC but NCP did not actually support it (`wpantund` did handle SLAAC if was needed by application). 

My main intention with this PR is to allow us to control SLAAC behavior and make it consistent with original model (e.g., have a way to disable SLAAC fully on an MTD build to save memory/code size). 

For this change, I earlier considered if we can somehow have SLAAC be enabled by default for CLI builds but not for NCP builds (to be consistent with the original model). But since this feature is now part of core, we need to decide if to enable/disable the feature when building core ot library (which can then be linked (shared) in both NCP and CLI builds). So no easy/clean way to add such a condition.

At the end, I went with the above model: i.e., disable by default and for it to be explicitly enabled if needed. I think it's safer to be explicit about support of SLAAC and as you mentioned, it follows all other features in OT. We explicitly enable it in all example makefiles.

Reply:
Ok thanks for explanation :+1: